core/src/main/java/org/infinispan/cache/impl/DecoratedCache.java

```diff
@@ -513,41 +495,41 @@ public V replace(K key, V value) {
 
    @Override
    public void addListener(Object listener) {
-      cacheImplementation.notifier.addListener(listener, classLoader.get());
+      cacheImplementation.notifier.addListener(listener, (ClassLoader) null);
    }
 
    @Override
    public void addListener(Object listener, KeyFilter<? super K> filter) {
-      cacheImplementation.notifier.addListener(listener, filter, classLoader.get());
+      cacheImplementation.notifier.addListener(listener, filter, null);
```

Comment:
This is public API, so it can't be removed right away. However, `ClassLoaderAwareListenable` and `ClassLoaderAwareFilteringListenable` should be deprecated.